src/api/java/com/minecolonies/api/inventory/container/ContainerCrafting.java

```diff
@@ -365,4 +375,38 @@ public BlockPos getPos()
     {
         return pos;
     }
+
+    /**
+     * Get for the remaining items. 
+     * @return
+     */
+    public java.util.List<ItemStack> getRemainingItems()
+    {
+        final Optional<ICraftingRecipe> iRecipe = this.world.getRecipeManager().getRecipe(IRecipeType.CRAFTING, craftMatrix, world);
+        RecipeBook r;
+        if(world.isRemote)
+        {
+            r = new ClientRecipeBook(this.world.getRecipeManager());
```

Comment:
this class might not load because it has the ClientRecipeBook in SMP. have you tested this in smp?

Reply:
what do we need it for? Just do check if its unlocked? I don't think we should care about that

Reply:
That's all it's used for. We probably don't need most of the checks there.

Reply:
yeah, remove it then. too error prone having this here.

Reply:
Gone